File: threedollar-api-user-service/src/e2e-test/java/com/depromeet/threedollar/api/userservice/SetupControllerTest.java
package com.depromeet.threedollar.api.userservice;

import org.junit.jupiter.api.AfterEach;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.context.TestConstructor;
import org.springframework.test.web.servlet.MockMvc;

import com.depromeet.threedollar.domain.rds.domain.lib.DatabaseCleaner;
import com.fasterxml.jackson.databind.ObjectMapper;

@TestConstructor(autowireMode = TestConstructor.AutowireMode.ALL)
@AutoConfigureMockMvc
@SpringBootTest
public abstract class SetupControllerTest {

    @Autowired
    protected MockMvc mockMvc;

    @Autowired
    protected ObjectMapper objectMapper;

    @Autowired
    protected DatabaseCleaner databaseCleaner;

    @AfterEach
    void cleanup() {
        databaseCleaner.cleanUp();
    }

}

File: threedollar-api-boss-service/src/e2e-test/kotlin/com/depromeet/threedollar/api/bossservice/SetupControllerTest.kt
package com.depromeet.threedollar.api.bossservice

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.context.TestConstructor
import org.springframework.test.web.servlet.MockMvc
import com.fasterxml.jackson.databind.ObjectMapper

@TestConstructor(autowireMode = TestConstructor.AutowireMode.ALL)
@AutoConfigureMockMvc
@SpringBootTest
internal abstract class SetupControllerTest : IntegrationTest() {

    @Autowired
    protected lateinit var objectMapper: ObjectMapper

    @Autowired
    protected lateinit var mockMvc: MockMvc

}
